src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8131,6 +8131,29 @@ public static string GetTarget(PSObject instance)
             return null;
         }
 
+        /// <summary>
+        /// Gets the target for a given file or directory, resolving symbolic links.
+        /// </summary>
+        /// <param name="instance">The FileInfo or DirectoryInfo type.</param>
+        /// <returns>The file path the instance points to.</returns>
+        public static string ResolvedTarget(PSObject instance)
+        {
+            if (instance.BaseObject is FileSystemInfo fileSysInfo)
+            {
+                try
+                {
+                    FileSystemInfo linkTarget = fileSysInfo.ResolveLinkTarget(true);
+                    return linkTarget is null ? fileSysInfo.FullName : linkTarget.FullName;
+                }
+                catch (IOException)
+                {
+                    return fileSysInfo.FullName;
+                }
```

Comment:
@hammy3502 As we discussed in https://github.com/PowerShell/PowerShell/pull/16490#discussion_r777764971, I think we don't need the try/catch block anymore given that `ResolveLinkTarget(true)` does not throw exception on a portable drive with `exFAT` file system (`ResolveLinkTarget(false)` throws but `ResolveLinkTarget(true)` doesn't).

Quoted it here again:
> So, maybe we can do the following:
>1. if `ResolveLinkTarget(true)` returns `non-null` value, just return that value. (I see you check on `symbTarget.Exists` here, but could it even be possible that the method returns a non-null value, but it doesn't exist?)
>2. if `ResolveLinkTarget(true)` returns `null` value, assuming that means it's not a link, and thus return the `FullName` or the current `FileSystemInfo` depending on what we want for return type.
>3. if `ResolveLinkTarget(true)` throws, then we assume that means it could be link, but something is wrong when resolving it. We just let the exception propagate in this case.

Even if `linkTarget` points to a non-existing item (somehow the target gets deleted), we should just return the full path to the target. Since .NET doesn't check on the existence of the target item before returning, we can follow the same behavior. So, the code within the try block looks good to me. I think you just need to remove the try/catch block.